openapi/renderer.go

```diff
@@ -0,0 +1,89 @@
+package openapi
+
+import (
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"path/filepath"
+	"strings"
+)
+
+type Renderer interface {
+	sanitiseName(string) string
+	sanitiseType(*Schema) string
+	getOutputFile(Traversable) string
+}
+
+var _ Renderer = GinRenderer{}
+
+type GinRenderer struct{}
+
+func (g GinRenderer) sanitiseType(s *Schema) string {
+	switch s.Type {
+	case "boolean":
+		return "bool"
+	case "array":
+		return "[]" + s.Items.GetName()
+	case "integer":
+		return "int"
+	case "object":
+		return "struct"
+	default:
+		return s.Type
```

Comment:
Should this not be exhaustive in the long run? Passing through types indiscriminately could break generated code without warning

Reply:
List already is exhaustive as far as I can tell, but there's no way to tell the compiler that other than returning some default, an error, or panicking.

I'd be happy to update the approach if we can find a counter example where this breaks. Can we do that as an issue of its own though?